app/services/ClaimToAdjustService.scala

```diff
@@ -65,6 +62,38 @@ class ClaimToAdjustService @Inject()(val financialDetailsConnector: FinancialDet
     }
   }
 
+  def getPoaAdjustmentReason(nino: Nino)(implicit hc: HeaderCarrier, user: MtdItUser[_], ec: ExecutionContext): Future[Either[Throwable, Option[String]]] = {
+    getPoaForNonCrystallisedFinancialDetails(nino).flatMap {
+      case Right(Some(financialDetails)) =>
+        getChargeHistory(financialDetails.documentDetails, financialDetailsConnector) map {
+          case Right(Some(chargeHistory)) => Right(chargeHistory.poaAdjustmentReason)
+          case Right(None) => Right(None)
+          case Left(ex) => Left(ex)
+        }
+      case Right(None) => Future.successful(Right(None))
+      case Left(ex) => Future.successful(Left(ex))
+    }
+  }
+
+  def getEnterPoAAmountViewModel(nino: Nino)(implicit hc: HeaderCarrier, user: MtdItUser[_], ec: ExecutionContext): Future[Either[Throwable, PoAAmountViewModel]] = {
+    for {
```

Comment:
line 80 and 81 would call getPoaForNonCrystallisedFinancialDetails;
essentially this is duplicate call to the API we should avoid, if you can leave TODO to address this ? in the worst case we can move it into tech debt